### DISCUSSION TITLE
"fields" for  realtime data

### DISCUSSION THREAD

**Discussion by denisgurev** [pocketbase/pocketbase#3659] [Q&A] [answered]
Is it possible with the js sdk or hooks to specify which fields to return when subscribing to * of a collection?

My use case is that I have a  large content field but I don't want to return it as it is very slow when testing with a mobile internet. When using fields with getList it works better but I want to also to be notified in realtime when a status change to update a label in my Svelte app.

**Comment by ganigeorgiev** [comment, marked as answer]
`filter`, `fields` (including the `:excerpt` modifier), `expand`, custom query and header parameters support is already implemented but it is currently available only in the `develop` branch (_it will be part of v0.20.0 but there is no ETA yet_).

If you want to test it, you can try building the `develop` branch on your own or download the [v0.20.0-rc prerelease](https://github.com/pocketbase/pocketbase/releases/tag/v0.20.0-rc). Note that you'll need also to update your JS SDK to the 0.20.0-rc next tag in your `package.json`.

```js
// JS SDK v0.20.0-rc
pb.collection("example").subscribe("*", (e) => {
  ...
}, {
  expand: "someRelField",
  filter: "status = 'active'",
  fields: "id,content:excerpt(100),expand.someRelField.*",
})
```

Alternatively, you can also change manually the realtime event message with the [`onRealtimeBeforeMessageSend` hook](https://pocketbase.io/docs/js-event-hooks/#onrealtimebeforemessagesend).

**Reply by denisgurev** [reply]
Thank you! This is exactly what I was looking for, I somehow missed the release, sorry.
